extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -620,17 +629,20 @@ public void runWith(Runnable task, Object context) {
                 // Only do context handling if it's non-null
                 if (context != null && context != currentContext) {
                     ContextInternal vertxContext = (ContextInternal) context;
-                    // The CDI contexts must not be propagated
-                    // First test if VertxCurrentContextFactory is actually used
-                    if (currentContextFactory != null) {
-                        List<String> keys = currentContextFactory.keys();
+                    if (!ignoredKeysSuppliers.isEmpty()) {
                         ConcurrentMap<Object, Object> local = vertxContext.localContextData();
-                        if (containsScopeKey(keys, local)) {
-                            // Duplicate the context, copy the data, remove the request context
-                            vertxContext = vertxContext.duplicate();
-                            vertxContext.localContextData().putAll(local);
-                            keys.forEach(vertxContext.localContextData()::remove);
-                            VertxContextSafetyToggle.setContextSafe(vertxContext, true);
+                        ContextInternal duplicateVertxContext = null;
+                        for (Supplier<List<String>> ignoredKeysSupplier : ignoredKeysSuppliers) {
```

Comment:
I'd suggest to build an aggregate `List<String> keys` first instead of iterating over all suppliers every time the context is captured...

Reply:
Supplier of `List<String>` kept original behavior which requested keys on every call. Looking at `io.quarkus.vertx.runtime.VertxCurrentContextFactory` keys are unmodifiable so I'll change this to list instead. Good point. Thanks